File: GTAHybrid/RN/RNBridge.m
//
//  RNBridge.m
//  GTAHybrid
//
//  Created by goingta on 2018/12/14.
//  Copyright © 2018 goingta. All rights reserved.
//

#import <React/RCTBridgeModule.h>
#import <React/RCTLog.h>

@interface RCT_EXTERN_MODULE(RNBridge, NSObject) //RCT_EXTERN_MODULE将模块导出到Reac-Native

- (dispatch_queue_t)methodQueue
{
    return dispatch_get_main_queue();
}

+ (BOOL)requiresMainQueueSetup
{
    return YES;
}

RCT_EXTERN_METHOD(command:(NSDictionary *)params)  //RCT_EXTERN_METHOD将方法导出到ReacNative

RCT_EXTERN_METHOD(commandExtension:(NSDictionary *)params callback:(RCTResponseSenderBlock)callback)  //RCT_EXTERN_METHOD将方法导出到ReacNative

@end

File: GTAHybrid/RN/RNBridge.swift
//
//  RNBridge.swift
//  GTAHybrid
//
//  Created by goingta on 2018/12/14.
//  Copyright © 2018 goingta. All rights reserved.
//

import Foundation

@objc(RNBridge) //@objc关键字,用于将方法或对象暴露给OC,以供OC调用访问
class RNBridge: NSObject {
    static let shared: RNBridge = RNBridge()
    
    override init() {
        
    }
    
    /// 和业务有关的命令扩展
    ///
    /// - Parameter command: 命令对象
    @objc public func command(_ params: [String: AnyObject]) -> Void {
        print(params)
    }
}
